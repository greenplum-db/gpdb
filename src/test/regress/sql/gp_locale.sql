DROP DATABASE IF EXISTS testdb3;
CREATE DATABASE testdb3 WITH LC_COLLATE='C' LC_CTYPE='C' TEMPLATE=template0;
\c testdb3

CREATE TABLE hi_안녕세계 (a int, 안녕세계 text) DISTRIBUTED BY (a);
INSERT INTO hi_안녕세계 VALUES(1, '안녕세계 1');
INSERT INTO hi_안녕세계 VALUES(42, '안녕세계 42');
SET optimizer_trace_fallback=on;

-- without setting optimizer_locale_for_query_to_dxl_translation the following queries should fall back.
SHOW optimizer_locale_for_query_to_dxl_translation;
EXPLAIN DELETE FROM hi_안녕세계 WHERE a=42;
EXPLAIN SELECT * FROM hi_안녕세계;

-- with valid optimizer_locale_for_query_to_dxl_translation setting the following queries shouldn't fall back.
SET optimizer_locale_for_query_to_dxl_translation='en_US.UTF-8';
EXPLAIN DELETE FROM hi_안녕세계 WHERE a=42;
DELETE FROM hi_안녕세계 WHERE a=42;

EXPLAIN SELECT * FROM hi_안녕세계;
SELECT * FROM hi_안녕세계;

RESET optimizer_locale_for_query_to_dxl_translation;
