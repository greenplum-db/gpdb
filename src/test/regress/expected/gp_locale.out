DROP DATABASE IF EXISTS testdb3;
CREATE DATABASE testdb3 WITH LC_COLLATE='C' LC_CTYPE='C' TEMPLATE=template0;
\c testdb3
CREATE TABLE hi_안녕세계 (a int, 안녕세계 text) DISTRIBUTED BY (a);
INSERT INTO hi_안녕세계 VALUES(1, '안녕세계 1');
INSERT INTO hi_안녕세계 VALUES(42, '안녕세계 42');
SET optimizer_trace_fallback=on;
-- without setting optimizer_locale_for_query_to_dxl_translation the following queries should fall back.
SHOW optimizer_locale_for_query_to_dxl_translation;
 optimizer_locale_for_query_to_dxl_translation 
-----------------------------------------------
 
(1 row)

EXPLAIN DELETE FROM hi_안녕세계 WHERE a=42;
                              QUERY PLAN                               
-----------------------------------------------------------------------
 Delete on "hi_안녕세계"  (cost=0.00..240.67 rows=17 width=10)
   ->  Seq Scan on "hi_안녕세계"  (cost=0.00..240.67 rows=17 width=10)
         Filter: (a = 42)
 Optimizer: Postgres-based planner
(4 rows)

EXPLAIN SELECT * FROM hi_안녕세계;
                                    QUERY PLAN                                     
-----------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..860.67 rows=49600 width=36)
   ->  Seq Scan on "hi_안녕세계"  (cost=0.00..199.33 rows=16533 width=36)
 Optimizer: Postgres-based planner
(3 rows)

-- with valid optimizer_locale_for_query_to_dxl_translation setting the following queries shouldn't fall back.
SET optimizer_locale_for_query_to_dxl_translation='en_US.UTF-8';
EXPLAIN DELETE FROM hi_안녕세계 WHERE a=42;
                              QUERY PLAN                               
-----------------------------------------------------------------------
 Delete on "hi_안녕세계"  (cost=0.00..240.67 rows=17 width=10)
   ->  Seq Scan on "hi_안녕세계"  (cost=0.00..240.67 rows=17 width=10)
         Filter: (a = 42)
 Optimizer: Postgres-based planner
(4 rows)

DELETE FROM hi_안녕세계 WHERE a=42;
EXPLAIN SELECT * FROM hi_안녕세계;
                                    QUERY PLAN                                     
-----------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..860.67 rows=49600 width=36)
   ->  Seq Scan on "hi_안녕세계"  (cost=0.00..199.33 rows=16533 width=36)
 Optimizer: Postgres-based planner
(3 rows)

SELECT * FROM hi_안녕세계;
 a |  안녕세계  
---+------------
 1 | 안녕세계 1
(1 row)

RESET optimizer_locale_for_query_to_dxl_translation;
